testing/test_adaptative.py

```diff
@@ -0,0 +1,157 @@
+        mt_transforms.CenterCrop2D(size=[48, 48]),
+        mt_transforms.ToTensor()
+    ]
+    train_transform = transforms.Compose(training_transform_list)
+
+    dataset = adaptative.HDF5Dataset(root_dir=PATH_BIDS,
+                                     subject_lst=train_lst,
+                                     hdf5_name='testing_data/mytestfile.hdf5',
+                                     csv_name='testing_data/hdf5.csv',
+                                     target_suffix="_lesion-manual",
+                                     contrast_lst=['T1w', 'T2w', 'T2star'],
+                                     ram=False,
+                                     contrast_balance={},
+                                     slice_axis=2,
+                                     transform=train_transform,
+                                     metadata_choice=False,
+                                     dim=2,
+                                     slice_filter_fn=SliceFilter(filter_empty_input=True, filter_empty_mask=True),
+                                     canonical=True,
+                                     roi_suffix="_seg-manual",
+                                     target_lst=['T2w'],
+                                     roi_lst=['T2w'])
+
+    dataset.load_into_ram(['T1w', 'T2w', 'T2star'])
+    print("Dataset RAM status:")
+    print(dataset.status)
+    print("In memory Dataframe:")
+    print(dataset.dataframe)
+    print('\n[INFO]: Test passed successfully. ')
+
+    print("\n[INFO]: Starting loader test ...")
+
+    device = torch.device("cuda:0" if torch.cuda.is_available() else "cpu")
+    cuda_available = torch.cuda.is_available()
+    if cuda_available:
+        torch.cuda.set_device(GPU_NUMBER)
+        print("Using GPU number {}".format(GPU_NUMBER))
+
+    training_transform_list = [
+        ivadomed_transforms.Resample(wspace=0.75, hspace=0.75),
+        ivadomed_transforms.ROICrop2D(size=[48, 48]),
+        mt_transforms.ToTensor()
+    ]
+    train_transform = transforms.Compose(training_transform_list)
+
+    train_loader = DataLoader(dataset, batch_size=BATCH_SIZE,
+                              shuffle=False, pin_memory=True,
+                              collate_fn=mt_datasets.mt_collate,
+                              num_workers=1)
+    model = models.HeMISUnet(modalities=['T1w', 'T2w', 'T2star'],
+                            depth=2,
+                            drop_rate=DROPOUT,
+                            bn_momentum=BN)
+
+    if cuda_available:
+        model.cuda()
+
+    step_scheduler_batch = False
+    optimizer = optim.Adam(model.parameters(), lr=INIT_LR)
+    scheduler = optim.lr_scheduler.CosineAnnealingLR(optimizer, N_EPOCHS)
+
+    load_lst, pred_lst, opt_lst, schedul_lst, init_lst, gen_lst = [], [], [], [], [], []
+    for epoch in tqdm(range(1, N_EPOCHS + 1), desc="Training"):
+        start_time = time.time()
+
+        start_init = time.time()
+        lr = scheduler.get_lr()[0]
+        model.train()
```

Comment:
Do you plan to feed the model in this testing function?

Reply:
Finally, I decided to test the training of HeMIS in `test_HeMIS.py` and forgot to remove those lines of code. Thx